src/gsCore/gsDofMapper.cpp

```diff
@@ -282,6 +295,42 @@ void gsDofMapper::setIdentity(index_t nPatches, size_t nDofs)
     m_dofs.resize( m_numFreeDofs, 0);
 }
 
+void gsDofMapper::permuteFreeDofs(const gsVector<index_t>& permutation)
+{
+    GISMO_ASSERT(m_curElimId==0, "finalize() was not called on gsDofMapper");
+    GISMO_ASSERT(m_numFreeDofs == permutation.size(), "permutation size does not match number of free dofs");
+    //GISMO_ASSERT(m_tagged.empty(), "you cannot permute the dofVector twice, combine the permutation");
+    //We could check here, that permutation is indeed a permutation
+
+    //make a copy of the old ordering, easiest way to implement the permutation. Inplace reordering is quite hard.
+    std::vector<index_t> dofs = m_dofs;
+    //first use a tempory coupled vector, other wise is_coupled_index() would not work correctly
+    std::vector<index_t> tagged;
+    tagged.reserve(taggedSize()); //reserve enough memory
+    
+    for(index_t i=0; i<(index_t)dofs.size();++i)
+    {
+        index_t idx = dofs[i];
+        if(is_free_index(idx))
+        {
+            m_dofs[i] = permutation[idx];
+            //fill  bookkeeping for tagged dofs
+            if(is_tagged_index(idx))
+                tagged.push_back(m_dofs[i]);
```

Comment:
There is a small bug here for all tagged indices which are not free
(they should stay in "m_tagged" upermuted)

Reply:
Just add "else if is_tag_index(idx) tagged.push_back(idx);" to fix it

Reply:
uhh, thats fishy bug. nice.